cassandra/cluster.py

```diff
@@ -1975,6 +1975,12 @@ def __init__(self, cluster, hosts, keyspace=None):
         while futures.not_done and not any(f.result() for f in futures.done):
             futures = wait_futures(futures.not_done, return_when=FIRST_COMPLETED)
 
+        if not any(f.result() for f in self._initial_connect_futures):
```

Comment:
Are there use cases for passing no hosts? If so, maybe this should be `if (not self._inital_connect_futures) and (not any(...)):`. I think that'd also make the tests require fewer changes, since empty `hosts` lists would be allowed.